src/Oro/Bundle/ActionBundle/Configuration/ConfigurationProvider.php

```diff
@@ -112,4 +150,28 @@ protected function resolveConfiguration(Collection $errors = null)
 
         return $data;
     }
+
+    /**
+     * @param ContainerBuilder|null $containerBuilder
+     * @return array
+     */
+    protected function loadConfiguration(ContainerBuilder $containerBuilder = null)
```

Comment:
This code should be placed a separate service to avoid a injection of parameter bar, make easy for reuse and add local caching